### PR TITLE
NOISSUE - Fix subtopic regex and restrict empty subtopic parts

### PR DESCRIPTION
**Pull request by anovakovic01**
### What does this do?
It restricts publish and subscribe to subtopic that contains `*` or `>` inside topic part.

### Have you included tests for your changes?
No.

### Did you document any new/modified functionality?
No.

### CODE REVIEW

**Review by manuio — DISMISSED** [review]

File: coap/api/transport.go [hunk 1]
```diff
@@ -112,14 +116,30 @@ func authorize(msg *gocoap.Message, res *gocoap.Message, cid string) (string, er
...
+	subtopic = strings.Replace(subtopic, "/", ".", -1)
+
+	elems := strings.Split(subtopic, ".")
+	filteredElems := []string{}
+	for _, elem := range elems {
+		if elem == "" {
+			continue
+		}
+
+		if len(elem) > 1 && (strings.Contains(elem, "*") || strings.Contains(elem, ">")) {
```
Comment (drasko) [inline coap/api/transport.go:133]: Why do we exclude joker topics?

**Review by drasko — CHANGES_REQUESTED** [review]

File: coap/api/transport.go (hunk 1 above)
Comment (anovakovic01) [reply coap/api/transport.go:133]: We didn't, we just exclude invalid ones, like `fo*o`.

**Review by nmarcetic — APPROVED** [review]

**Comment by mteodor** [comment]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**